android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -151,6 +153,21 @@ public void onReceive(Context context, Intent intent) {
     getReactApplicationContext().registerReceiver(headphoneConnectionReceiver, filter);
   }
 
+  private void initializeLocationServicesEnabledReceiver() {
+    IntentFilter filter = new IntentFilter();
+    filter.addAction(LocationManager.MODE_CHANGED_ACTION);
+
+    locationServicesEnabledReceiver = new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        boolean isEnabled = isLocationEnabledSync();
+        sendEvent(getReactApplicationContext(), "RNDeviceInfo_locationEnabledDidChange", isEnabled);
+      }
+    };
+
+    getReactApplicationContext().registerReceiver(locationServicesEnabledReceiver, filter);
+  }
+
 
   @Override
   public void onCatalystInstanceDestroy() {
```

Comment:
should it stop listening on catalyst destroy? should probably also de-register if there are no listeners